### PR TITLE
STITCH-2218/STITCH-2219 Swift SDK: Conflict handling in the data synchronizer - R2L/L2R

### PR DESCRIPTION
**Pull request by jsflax**
Contains both tickets and the beginning few integration tests.

### CODE REVIEW

File: Core/Services/StitchCoreRemoteMongoDBService/Sources/StitchCoreRemoteMongoDBService/Sync/DataSynchronizer.swift [hunk 1]
```diff
@@ -191,11 +191,988 @@ public class DataSynchronizer: NetworkStateDelegate, FatalErrorListener {
...
+
+        logger.i(
+            "t='\(logicalT)': resolveConflict ns=\(namespace) documentId=\(documentId) resolving conflict between localOp=\(uncommittedChangeEvent.operationType) "
+                + "remoteOp=\(remoteEvent.operationType)")
+
+        // 2. Based on the result of the handler determine the next state of the document.
+        let resolvedDocument = try DataSynchronizer.resolveConflictWithResolver(
+            conflictResolver: conflictHandler,
+            documentId: documentId,
+            localEvent: uncommittedChangeEvent,
```
[the remark this answers is not in the document]
Comment (adamchel) [reply Core/Services/StitchCoreRemoteMongoDBService/Sources/StitchCoreRemoteMongoDBService/Sync/DataSynchronizer.swift]: nvm

File: Core/Services/StitchCoreRemoteMongoDBService/Sources/StitchCoreRemoteMongoDBService/Sync/NamespaceChangeStreamDelegate.swift [hunk 2]
```diff
@@ -25,12 +35,17 @@ class NamespaceChangeStreamDelegate: SSEStreamDelegate, NetworkStateDelegate {
...
         self.authMonitor = authMonitor
+        super.init()
+        networkMonitor.add(networkStateDelegate: self)
     }
 
     /**
      Open the event stream.
      */
     func start() throws {
+        objc_sync_enter(self)
```
[the remark this answers is not in the document]
Comment (jsflax) [reply Core/Services/StitchCoreRemoteMongoDBService/Sources/StitchCoreRemoteMongoDBService/Sync/NamespaceChangeStreamDelegate.swift]: Will be handled in RWLock ticket coming later.

File: Core/StitchCoreSDK/Sources/StitchCoreSDK/Internal/Net/FoundationHTTPSSEStream.swift [hunk 3]
```diff
@@ -1,8 +1,9 @@
 import Foundation
 
 internal class FoundationURLSessionDataDelegate: NSObject, URLSessionDataDelegate {
-    private weak var stream: FoundationHTTPSSEStream?
+    fileprivate weak var stream: FoundationHTTPSSEStream?
     fileprivate weak var session: URLSession?
+    fileprivate var fulfillClose: Bool = false
```
[the remark this answers is not in the document]
Comment (jsflax) [reply Core/StitchCoreSDK/Sources/StitchCoreSDK/Internal/Net/FoundationHTTPSSEStream.swift]: I can add one for safety, just in case.

**Comment by jsflax** [comment]
@adamchel PTAL

**Pull request merged**